### PR TITLE
Update REAMDE to reference Microsoft Learn

### PR DESCRIPTION
**Pull request by zhiyuanliang-ms**
## Why this PR?

The content of README has been migrated to the Microsoft Learn: https://learn.microsoft.com/en-us/azure/azure-app-configuration/feature-management-dotnet-reference?pivots=stable-version

The README in this repo should reference Microsoft Learn.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,883 +1,20 @@
...
-services.AddSingleton<IFeatureDefinitionProvider, InMemoryFeatureDefinitionProvider>()
-        .AddFeatureManagement()
-```
+* [.NET Console App](./examples/ConsoleApp)
+* [.NET Console App with Targeting](./examples/TargetingConsoleApp)
+* [ASP.NET Core Web App (Razor Page)](./examples/RazorPages)
+* [ASP.NET Core Web App (MVC)](./examples/FeatureFlagDemo)
+* [Blazor Server App](./examples/BlazorServerApp)
 
 # Contributing
```
Comment (zhenlan) [inline README.md:19]: Can we change this to 2 #?

File: README.md [hunk 2]
```diff
@@ -1,883 +1,20 @@
...
-* Feature Flag lifetime management
-  * Configuration values can change in real-time, feature flags can be consistent across the entire request
-* Simple to Complex Scenarios Covered
-  * Toggle on/off features through declarative configuration file
-  * Dynamically evaluate state of feature based on call to server
-* API extensions for ASP.NET Core and MVC framework
-  * Routing
-  * Filters
-  * Action Attributes
+Official documentation on how to use feature management is available in the [Microsoft Learn](https://learn.microsoft.com/azure/azure-app-configuration/feature-management-dotnet-reference).
```
Comment (zhenlan) [inline README.md:7]: This sentence can be moved to the end of the first paragraph.

File: README.md [hunk 3]
```diff
@@ -1,883 +1,20 @@
...
 
 **API Reference**: https://go.microsoft.com/fwlink/?linkid=2091700
```
Comment (zhenlan) [inline README.md:9]: We don't' need this anymore.

**Review by zhenlan — APPROVED** [review]

**Comment by rossgrambo** [comment]
I think we should offer a slightly more explicit link to the docs like in the [ASP.NET readme](https://github.com/dotnet/aspnetcore?tab=readme-ov-file#get-started)

They have the same descriptive paragraph with a link at the end- but then they have a quick bullet list with some main doc urls. Even if it's just something like:

* [.NET Feature Management Overview](https://learn.microsoft.com/en-us/azure/azure-app-configuration/feature-management-dotnet-reference?pivots=stable-version)
* [API Reference](https://learn.microsoft.com/en-us/dotnet/api/overview/azure/appconfiguration/featuremanagement?view=azure-dotnet)

The customer I have in mind is someone who's familiar with FM but wants to get to a specific feature- like variants. The paragraph isn't relevant to them and the current link seems like it's a part of the intro.

**Comment by zhiyuanliang-ms** [comment]
@rossgrambo 

>  they have a quick bullet list with some main doc urls. Even if it's just something like:

Do you mean the `Get started` section in the [ASP.NET readme](https://github.com/dotnet/aspnetcore?tab=readme-ov-file#get-started)

How about this: we also add a `Get started` section in our readme. We will list the following topics: 
`Concepts`: https://learn.microsoft.com/en-us/azure/azure-app-configuration/concept-feature-management (This page needs to be updated. It should reference the new .NET feature management page)
`feature flag configuration`: https://learn.microsoft.com/en-us/azure/azure-app-configuration/feature-management-dotnet-reference?pivots=stable-version#feature-flag-configuration
`How to use feature management`: https://learn.microsoft.com/en-us/azure/azure-app-configuration/feature-management-dotnet-reference?pivots=stable-version#consumption

**Comment by zhiyuanliang-ms** [comment]
I have a question in my mind: who will read this README? Newcomers who just find this feature management library or pro users who have been using the library and want to have a deeper understand of this repo? For the newcomers, we can write a simple example in the readme and give some links to the Microsoft Learn (we have plenty of quickstarts there).
For the pro users, I think a detailed index will be helpful. How about keep the original index section here:

```
## Index
* [Feature Flags](#feature-flags)
    * [Feature Filters](#feature-filters)
    * [Feature Flag Declaration](#feature-flag-declaration)
* [Consumption](#consumption)
* [ASP.NET Core Integration](#ASPNET-core-integration)
* [Implement a Feature Filter](#implementing-a-feature-filter)
* [Providing a Context For Feature Evaluation](#providing-a-context-for-feature-evaluation)
* [Built-in Feature Filters](#built-in-feature-filters)
* [Targeting](#targeting)
  * [Targeting Exclusion](#targeting-exclusion)
* [Caching](#caching)
* [Custom Feature Providers](#custom-feature-providers)
```

**Comment by zhenlan** [comment]
> How about keep the original index section here

I would not duplicate that content here. It adds extra burden for maintenance.

If we like, we can have something like this:

_Follow the [quickstart]() guide to learn how to integrate feature flags from Azure App Configuration into your .NET applications. Check out the [feature reference]() document for a full feature rundown. See the [API reference]() for more information._

or shorter version:

_Follow the [quickstart]() for .NET feature flag integration. View the [feature reference]() and [API reference]() for further details._

(The links are placeholders only)

**Review by jimmyca15 — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -1,885 +1,24 @@
...
 
-[![Microsoft.FeatureManagement](https://img.shields.io/nuget/v/Microsoft.FeatureManagement?label=Microsoft.FeatureManagement)](https://www.nuget.org/packages/Microsoft.FeatureManagement)
-[![Microsoft.FeatureManagement.AspNetCore](https://img.shields.io/nuget/v/Microsoft.FeatureManagement.AspNetCore?label=Microsoft.FeatureManagement.AspNetCore)](https://www.nuget.org/packages/Microsoft.FeatureManagement.AspNetCore)
+Click the following link to see the official documentation on how to use feature management:
 
-Feature flags provide a way for .NET and ASP.NET Core applications to turn features on or off dynamically. Developers can use feature flags in simple use cases like conditional statements to more advanced scenarios like conditionally adding routes or MVC filters. Feature flags are built on top of the .NET Core configuration system. Any .NET Core configuration provider is capable of acting as the backbone for feature flags.
+[***Official Documentation***](https://learn.microsoft.com/azure/azure-app-configuration/feature-management-dotnet-reference)
```
[the remark this answers is not in the document]
Comment (jimmyca15) [reply README.md:7]: Good point. I didn't realize that the same link was in the Get Started below. I don't think we should have both.

As to which one to keep, I don't like how the Get Started's feature reference link is embedded in a paragraph. I feel the navigation is easier when the link is stand alone and well named like the above example.

However, that could be said about all the links in Get Started. If it sounds reasonable one thing we could do is have no links in get started and instead enumerate the links afterwards like this.

---

# Get started

[Quickstart](https://learn.microsoft.com/azure/azure-app-configuration/quickstart-feature-flag-dotnet): A quickstart guide is available to learn how to integrate feature flags from Azure App Configuration into your .NET applications.

[Feature Reference](https://learn.microsoft.com/azure/azure-app-configuration/feature-management-dotnet-reference): This document provides a full feature rundown.

[API reference](https://go.microsoft.com/fwlink/?linkid=2091700): This API reference details the API surface of this library.

[the remark this answers is not in the document]
Comment (zhenlan) [reply README.md:7]: I like it.

File: README.md [hunk 5]
```diff
@@ -1,885 +1,24 @@
...
-* Feature Flag lifetime management
-  * Configuration values can change in real-time, feature flags can be consistent across the entire request
-* Simple to Complex Scenarios Covered
-  * Toggle on/off features through declarative configuration file
-  * Dynamically evaluate state of feature based on call to server
-* API extensions for ASP.NET Core and MVC framework
-  * Routing
-  * Filters
-  * Action Attributes
+[**API reference**](https://go.microsoft.com/fwlink/?linkid=2091700): This API reference details the API surface of this library.
```
Comment (zhenlan) [inline README.md:11]: We have more than one library. Something like this?

This API reference details the API surface of the libraries contained within this repository.

**Review by zhenlan — APPROVED** [review]

**Review by jimmyca15 — APPROVED** [review]

**Pull request merged**